pkg/common/cns-lib/vsphere/virtualcenter.go

```diff
@@ -141,7 +141,7 @@ type VirtualCenterConfig struct {
 }
 
 // NewClient creates a new govmomi Client instance.
-func (vc *VirtualCenter) NewClient(ctx context.Context) (*govmomi.Client, error) {
+func (vc *VirtualCenter) NewClient(ctx context.Context, useragent string) (*govmomi.Client, error) {
```

Comment:
Can't we calculate the useragent in this method instead of passing it as argument in the method?

Reply:
if we calculate useragent in NewClient() then we can't set unique useragent like we doing for list view
"k8s-csi-useragent-<cluster-id>-listview" so moved logic to other place where we calculate useragent and pass it to this function.